jenkins_pipelines/scripts/json_generator/ibs_osc_client.py

```diff
@@ -68,7 +69,7 @@ def _parse_embargo_date(self, text: str) -> date:
             try:
                 return datetime.strptime(text, format).date()
             except:
-                print(f"\tInvalid date format {format} for OBS:EmbargoDate value {text}")
+                logging.error(f"\tInvalid date format {format} for OBS:EmbargoDate value {text}")
         
         raise ValueError(f"Failed to parse OBS:EmbargoDate value {text} with all available date formats")
```

Comment:
You can just raise `ValueError` with that message instead, optional

Reply:
The logic of this function is to try and parse the text we get in the date field with all the available formats we know to be used.
If one works, cool, we return the parsed date. If it does not, I want to have a log of the failure but not raise an exception  - otherwise the function will stop at that point and other formats won't be tried.

If none of them works, then line 73 will raise the ValueError and lead to a crash. At this point the previous logs would help knowing what the text was and what date formats we tried.

Reply:
I missed it is `for` loop, thank you for clarification.

Resolving